partiql-parser/src/lalr/lexer.rs

```diff
@@ -475,32 +589,86 @@ mod tests {
+            toks.into_iter().map(|(_s, t, _e)| t).collect::<Vec<_>>()
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn select_comment_block() -> Result<(), LexicalError> {
+        let query = "SELECT /*comment*/ g";
+        let lexer = Lexer::new(query);
+        let toks: Vec<_> = lexer.collect::<Result<_, _>>()?;
+
+        assert_eq!(
+            vec![
+                Token::Select,
+                Token::CommentBlock("/*comment*/".to_owned()),
+                Token::Identifier("g".to_owned()),
+            ],
+            toks.into_iter().map(|(_s, t, _e)| t).collect::<Vec<_>>()
+        );
+        Ok(())
+    }
+
     #[test]
     fn err_invalid_input() {
         let query = "SELECT # FROM data GROUP BY a";
         let toks: Result<Vec<_>, LexicalError> = Lexer::new(query).collect();
         assert!(toks.is_err());
-        match toks.unwrap_err() {
-            LexicalError::InvalidInput((s, t, e)) => {
-                assert_eq!(t, "#");
-                assert_eq!(s, 7);
-                assert_eq!(e, 8);
-            }
-            _ => panic!("Error should be LexicalError::InvalidInput"),
-        }
+        let error = toks.unwrap_err();
+        assert_eq!(error.to_string(), r##"Lexing error: invalid input `#`"##);
+        assert!(matches!(error, LexicalError::InvalidInput((7, s, 8)) if s == "#"));
     }
 
     #[test]
     fn err_unterminated_ion() {
         let query = r#" ` "fooo` "#;
         let toks: Result<Vec<_>, LexicalError> = Lexer::new(query).collect();
         assert!(toks.is_err());
-        match toks.unwrap_err() {
-            LexicalError::UnterminatedIonLiteral((s, _t, e)) => {
-                assert_eq!(s, 1);
-                assert_eq!(e, 8);
-            }
-            _ => panic!("Error should be LexicalError::UnterminatedIonLiteral"),
-        }
+        let error = toks.unwrap_err();
+        assert!(matches!(
+            error,
+            LexicalError::UnterminatedIonLiteral((1, (), 9))
+        ));
+        assert_eq!(error.to_string(), "Lexing error: unterminated ion literal");
+    }
+
+    #[test]
+    fn err_unterminated_comment() {
+        let query = r#" /*12345678"#;
+        let toks: Result<Vec<_>, LexicalError> = Lexer::new(query).collect();
+        assert!(toks.is_err());
+        let error = toks.unwrap_err();
+        assert!(matches!(
+            error,
+            LexicalError::UnterminatedComment((1, (), 10))
+        ));
+        assert_eq!(error.to_string(), "Lexing error: unterminated comment");
+    }
+
+    #[test]
+    fn err_unterminated_ion_comment() {
+        let query = r#" `/*12345678`"#;
+        let toks: Result<Vec<_>, LexicalError> = EmbeddedIonLexer::new(query).collect();
+        assert!(toks.is_err());
+        let error = toks.unwrap_err();
+        assert!(matches!(
+            error,
+            LexicalError::UnterminatedComment((2, (), 11))
+        ));
+        assert_eq!(error.to_string(), "Lexing error: unterminated comment");
```

Comment:
Nice set of tests! I could only think of a couple more tests that could be added:
- query starting with block comment closing, `*/`
  - Something like: `*/ SELECT g`
- within an Ion literal, start with a block comment closing, `*/`
  - Something like: `` `*/` SELECT g``

Reply:
At least as currently designed, these are both valid lexes:

- `*/ SELECT g`
  - This is a `*` operator, then a `/` operator, then a `SELECT` keyword, then a `g` identifier
  - This might be worth adding a token for, so we can provide better ergonomics of error messaging, though...
- `` `*/` SELECT g ``
  -  This is an ion string, then a `SELECT` keyword, then a `g` identifier
     - (the ion string is technically invalid, but this lexer is just trying to find the text between the backticks that we'll have a proper ion library parse)

Reply:
This is interesting, at the end of the day machines just do their job :)

Maybe a TODO for the first item?

Reply:
Ok, got it. 1st case be a parser error then?